scripts/calendar/mabims_islamic_generator.py

```
@@ -0,0 +1,142 @@
+            # Month starts the next day
+            res = next_sunset.datetime().date() + timedelta(days=1)
+        else:
+            # Hilal not visible; month starts a day later (30 days completed)
+            res = next_sunset.datetime().date() + timedelta(days=2)
+        return res
+
+
+def run_generator():
+    engine = MabimsEngine()
+    h_start, h_end = HIJRI_RANGE
+    dates_map = defaultdict(lambda: defaultdict(list))
```

Ruff (0.15.11)

[error] Undefined name `defaultdict`

(F821)